Tools/PIDML/kaon_pid.cxx

```diff
@@ -0,0 +1,138 @@
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// This task produces invariant mass vs. momentum and dEdX in TPC vs. momentum
+/// for Kaons using ML PID from the PID ML ONNX Model.
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/runDataProcessing.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Multiplicity.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "CommonConstants/PhysicsConstants.h"
+#include "TMath.h"
+#include <math.h>
+#include "TLorentzVector.h"
+#include "TDatabasePDG.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Tools/PIDML/pidOnnxModel.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+namespace o2::aod
+{
+using MyCollisions = soa::Join<aod::Collisions,
+                               aod::EvSels,
+                               aod::Mults>;
+using MyTracks = soa::Join<aod::FullTracks, aod::TracksExtra, aod::pidTOFbeta,
+                           aod::TOFSignal, aod::TracksDCA>;
+using MyCollision = MyCollisions::iterator;
+using MyTrack = MyTracks::iterator;
+} // namespace o2::aod
+
+struct kaon_pid {
+  PidONNXModel* pidModel; // creates a pointer to a new instance 'pidmodel'.
+  HistogramRegistry histos{"Histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  Configurable<float> ConfZvtxCut{"ConfZvtxCut", 10, "Z vtx cut"};
+  Configurable<float> ConfEtaCut{"ConfEtaCut", 0.8, "Pseudorapidity cut"};
+  Configurable<float> ConfMaxPtCut{"ConfMaxPtCut", 3.0, "Max Pt cut"};
+  Configurable<float> ConfMinPtCut{"ConfMinPtCut", 0.5, "Min Pt cut"};
+  Configurable<float> ConfMinNSigmaTPCCut{"ConfMinNSigmaTPCCut", 3., "N-sigma TPC cut"};
+  Configurable<float> ConfChargeCut{"ConfChargeCut", 0., "N-sigma TPC cut"};
+  Configurable<std::string> cfgPathLocal{"local-path", "/home/shira/kaon-mlpid/", "base path to the local directory with ONNX models"};
+  Configurable<std::string> cfgPathCCDB{"ccdb-path", "Users/m/mkabus/PIDML", "base path to the CCDB directory with ONNX models"};
+  Configurable<std::string> cfgCCDBURL{"ccdb-url", "http://alice-ccdb.cern.ch", "URL of the CCDB repository"};
+  Configurable<int> CfgPid{"pid", 321, "PID to predict"};
+  Configurable<double> CfgCertainty{"certainty", 0.5, "Minimum certainty above which the model accepts a particular type of particle"};
+  Configurable<uint32_t> cfgDetector{"detector", kTPCTOFTRD, "What detectors to use: 0: TPC only, 1: TPC + TOF, 2: TPC + TOF + TRD"};
+  Configurable<uint32_t> cfgTimestamp{"timestamp", -8.0, "Fixed timestamp"};
```

Comment:
Put a realistic value, please. Or at least -1, the only one negative value possible for a timestamp.